apps/hannk/interpreter/model.cpp

```diff
@@ -142,7 +154,7 @@ void Tensor::allocate() {
         return;
     }
 
-    if (is_dynamic()) {
+    if (is_dynamic() || is_external()) {
         return;
```

Comment:
I was going to suggest asserting buffer_.data() is not null, but that would clearly fail if the if above doesn't return before getting here. Is this case actually hit?

Reply:
Yes; we mark Tensors as external (or not) before we call allocate().